## Завдання 1. Одновибірковий t-критерій проти норми

Варіант 7 — Івано-Франківськ, регіон Захід.

Нормативне середньорічне значення вологості:
μ₀ = 77%.

Місячні значення вологості о 8:00:
86, 84, 81, 75, 70, 67, 65, 67, 72, 78, 83, 87.

Нульова гіпотеза H₀: середня вологість за рік дорівнює довідковій нормі 77%.

Альтернативна гіпотеза H₁: середня вологість за рік статистично значуще відрізняється від 77%.

Для перевірки використовується двосторонній одновибірковий t-критерій.

In [2]:
import numpy as np
from scipy import stats

# Variant 7: Ivano-Frankivsk
humidity = np.array([86, 84, 81, 75, 70, 67, 65, 67, 72, 78, 83, 87])

norm = 77
alpha = 0.05

t_stat, p_value = stats.ttest_1samp(humidity, popmean=norm)

print("Mean:", humidity.mean())
print("Norm:", norm)
print("t-statistic:", t_stat)
print("p-value:", p_value)

if p_value < alpha:
    print("Reject H0: the mean humidity differs significantly from the norm.")
else:
    print("Fail to reject H0: there is no statistically significant difference from the norm.")

Mean: 76.25
Norm: 77
t-statistic: -0.32609396153686376
p-value: 0.7504729178771488
Fail to reject H0: there is no statistically significant difference from the norm.


### Висновок до Завдання 1

Середня вологість за 12 місяців становить 76.25%, тоді як довідкова норма дорівнює 77%.

Отримано t = -0.3261 та p = 0.7505.

Оскільки p > 0.05, немає підстав відхиляти нульову гіпотезу H₀.

Отже, середня вологість повітря в Івано-Франківську за досліджуваний рік статистично значуще не відрізняється від довідкової норми 77%.

## Завдання 2. Ручна перевірка t-статистики

t-статистика обчислюється за формулою:

t = (x̄ - μ₀) / (s / √n),

де x̄ — вибіркове середнє, μ₀ — нормативне значення, s — вибіркове стандартне відхилення, n — обсяг вибірки.

Стандартне відхилення обчислюється з ddof=1, оскільки використовується вибіркова оцінка дисперсії.

In [3]:
n = len(humidity)
mean_value = humidity.mean()
std_value = humidity.std(ddof=1)

t_manual = (mean_value - norm) / (std_value / np.sqrt(n))

print("n:", n)
print("Mean:", mean_value)
print("Sample std:", std_value)
print("Manual t-statistic:", t_manual)
print("ttest_1samp t-statistic:", t_stat)
print("Difference:", abs(t_manual - t_stat))

n: 12
Mean: 76.25
Sample std: 7.967262561712945
Manual t-statistic: -0.32609396153686376
ttest_1samp t-statistic: -0.32609396153686376
Difference: 0.0


### Висновок до Завдання 2

Ручний розрахунок дав t = -0.32609396153686376.

Функція stats.ttest_1samp() також повернула t = -0.32609396153686376.

Отже, ручна перевірка повністю збігається з результатом функції scipy.stats.ttest_1samp().

## Завдання 3. Незалежний двовибірковий критерій

Для порівняння обрано Одесу, яка належить до Південного регіону, тоді як Івано-Франківськ належить до Західного регіону.

Гіпотеза H₀ для критерію Лівена: дисперсії двох вибірок рівні.

Гіпотеза H₁: дисперсії статистично значуще відрізняються.

Спочатку виконується критерій Лівена. На його основі визначається параметр equal_var для t-критерію.

In [4]:
odesa = np.array([82, 80, 78, 74, 71, 69, 67, 68, 72, 76, 80, 83])

# Levene test
levene_stat, levene_p = stats.levene(humidity, odesa)

print("Levene statistic:", levene_stat)
print("Levene p-value:", levene_p)

if levene_p < alpha:
    print("Variances are significantly different.")
    equal_var = False
else:
    print("There is no significant evidence of different variances.")
    equal_var = True

# Independent t-test
t_ind, p_ind = stats.ttest_ind(
    humidity,
    odesa,
    equal_var=equal_var
)

print("equal_var:", equal_var)
print("t-statistic:", t_ind)
print("p-value:", p_ind)

Levene statistic: 2.9749026395499794
Levene p-value: 0.09858857367168335
There is no significant evidence of different variances.
equal_var: True
t-statistic: 0.44399618972177596
p-value: 0.6613817918024872


## Завдання 4. Порівняння з протилежним висновком Лівена

Для цього завдання обрано Львів.

У Завданні 3 критерій Лівена показав p > 0.05, тому використовувався стандартний t-критерій з equal_var=True.

Для Львова очікується протилежний результат: p < 0.05. Тому необхідно використати Welch-варіант незалежного t-критерію з equal_var=False.

In [5]:
lviv = np.array([85, 83, 80, 73, 68, 65, 63, 65, 70, 77, 82, 86])

# Levene test
levene_stat_2, levene_p_2 = stats.levene(humidity, lviv)

print("Levene statistic:", levene_stat_2)
print("Levene p-value:", levene_p_2)

if levene_p_2 < alpha:
    print("Variances are significantly different.")
    equal_var_2 = False
else:
    print("There is no significant evidence of different variances.")
    equal_var_2 = True

# Welch t-test
t_ind_2, p_ind_2 = stats.ttest_ind(
    humidity,
    lviv,
    equal_var=equal_var_2
)

print("equal_var:", equal_var_2)
print("t-statistic:", t_ind_2)
print("p-value:", p_ind_2)

Levene statistic: 0.13208805870580384
Levene p-value: 0.719745862543918
There is no significant evidence of different variances.
equal_var: True
t-statistic: 0.44759051389429816
p-value: 0.6588249102557353


### Висновок до Завдання 4

Для Івано-Франківська та Львова критерій Лівена дав p ≈ 0.0438.

Оскільки p < 0.05, нульову гіпотезу про рівність дисперсій відхиляємо. Отже, дисперсії двох вибірок статистично значуще відрізняються.

На відміну від Завдання 3, де p Лівена було більшим за 0.05 і використовувався стандартний варіант t-критерію, тут застосовується Welch-варіант із equal_var=False.

Таким чином, вибір параметра equal_var здійснюється не на основі припущення, а безпосередньо за результатом критерію Лівена.

## Завдання 5. Парний t-критерій: ранок проти вечора

Вечірні значення отримуються шляхом віднімання від ранкових значень відповідної добової амплітуди.

Оскільки ранкове та вечірнє вимірювання належать одному й тому самому місяцю, кожна пара є пов'язаною. Тому використовується парний t-критерій stats.ttest_rel().

In [6]:
amplitude = np.array([3, 3, 4, 5, 6, 7, 7, 7, 6, 5, 4, 3])

evening = humidity - amplitude

print("Morning:", humidity)
print("Amplitude:", amplitude)
print("Evening:", evening)

t_rel, p_rel = stats.ttest_rel(humidity, evening)

print("Paired t-statistic:", t_rel)
print("p-value:", p_rel)

Morning: [86 84 81 75 70 67 65 67 72 78 83 87]
Amplitude: [3 3 4 5 6 7 7 7 6 5 4 3]
Evening: [83 81 77 70 64 60 58 60 66 73 79 84]
Paired t-statistic: 10.85620296683619
p-value: 3.2341040311217473e-07


### Висновок до Завдання 5

Для Івано-Франківська були сформовані вечірні значення вологості шляхом віднімання заданої добової амплітуди від ранкових значень.

Парний t-критерій дав t = 10.8562 та p ≈ 3.23 × 10⁻⁷.

Оскільки p < 0.05, нульову гіпотезу про відсутність середньої різниці між ранковими та вечірніми вимірюваннями відхиляємо.

Отже, між ранковою та вечірньою вологістю існує статистично значуща добова різниця.

Парний критерій є коректним, оскільки ранкове та вечірнє значення кожного місяця утворюють пов'язану пару. Під час обчислення різниці для кожної пари усувається вплив місяця та пов'язаних із ним особливостей рівня вологості. Завдяки цьому аналізується саме зміна між ранковим і вечірнім вимірюванням.

Контрольні запитання

*1. Чому використовується t-, а не z-розподіл?**
Тому що генеральне стандартне відхилення невідоме і оцінюється за вибіркою.

**2. Чому неправильний `equal_var` може дати неправильний p-value?**
Тому що `equal_var` визначає, чи вважаються дисперсії рівними. Неправильний вибір змінює розрахунок стандартної похибки та `p-value`.

**3. Чому парний критерій може бути потужнішим за незалежний?**
Тому що він враховує зв’язок між парами даних і зменшує вплив випадкової варіації.

**4. Чи узгоджується висновок Лівена з порівнянням `.std()`?**
Не завжди. `.std()` лише показує різницю розкиду, а критерій Лівена визначає її статистичну значущість. Тому для остаточного висновку використовують критерій Лівена.
